docs/log-levels.md

```diff
@@ -50,7 +50,7 @@ Some libraries and situations may not be entirely clear with regards to what log
 
 - Could include various additional information about a request, such as various diagnostics about created data structures, the state of caches or similar, which are created in order to serve a request.
 - Could include "begin operation" and "end operation" logs.
-  - However, please consider using [swift-distributed-tracing](https://github.com/apple/swift-distributed-tracing) to instrument "begin" and "end" events, as tracing can be more efficient than logging. Logging may need to create string representations of the durations, log levels, and other fields.
+- However, please consider using [swift-distributed-tracing](https://github.com/apple/swift-distributed-tracing) to instrument "begin" and "end" events, as tracing can be more efficient than logging. Logging may need to create string representations of the durations, log levels, and other fields.
```

Comment:
> tracing can be more efficient than logging

Bit of a blanket statement there.  swift-distributed-tracing brings in things that can make stuff more complicated than logging, such as context propagation and the whole tracing model with trace/span/span events etc.

> Logging may need to create string representations of the durations, log levels, and other fields.

This doesn't mean logging is less efficient as a result.  Some binary formats are less efficient than JSON, for example.

Reply:
Yeah I'll reword that sentence - I did want to keep some shout out to tracing here but I'll reword it a bit. Basically, "once you start debugging durations and events and stuff, tracing may be helpful to you so check it out".

 It means to explain that it automatically propagates context and we're making changes in the language and concurrency runtime to make propagation trivial, it's been long time in the making: https://github.com/apple/swift-evolution/blob/main/proposals/0311-task-locals.md :-)

Reply:
I made it a sidenote at the end of the section

> > You may also want to consider using [swift-distributed-tracing](https://github.com/apple/swift-distributed-tracing) to instrument "begin" and "end" events, as tracing may give you additional insights into your system behavior you would have missed with just manually analysing log statements.